src/iosrv/init.c

```diff
@@ -498,7 +498,7 @@ server_init(int argc, char *argv[])
 	rc = modules_load();
 	if (rc)
 		/* Some modules may have been loaded successfully. */
-		D_GOTO(exit_mod_loaded, rc);
+		D_GOTO(exit_mod_init, rc);
```

Comment:
The comment on line 500 doesn't fit with this change.   It was calling exit_mod_init because some modules may have loaded successfully.  This will potentially cause reachable leaks (not a huge issue).  It seems like it would be better to fix ds_iv_fini so that it doesn't seg fault instead of not finalizing any of the successfully loaded modules.

Reply:
Well I can remove the comment.
I have tested the correct+early exit of daos_io_server with my patch, in case modules_load() fails ! 
In fact, ds_iv_fini() was failing because if ds_iv_init() has not been called, and thus ds_iv_ns_list list-head has not been initialized!
As far as I can see there is dss_module_cleanup_all() may be to undo modules_load() here ??!!...